pkg/egresspolicy/egresspolicy.go

```diff
@@ -0,0 +1,520 @@
+		}
+		dstCidrList = append(dstCidrList, cidr)
+	}
+
+	var endpointSelectorList []api.EndpointSelector
+
+	for _, egressRule := range cenp.Spec.Egress {
+		if egressRule.NamespaceSelector != nil {
+			prefixedNsSelector := egressRule.NamespaceSelector
+			matchLabels := map[string]string{}
+			// We use our own special label prefix for namespace metadata,
+			// thus we need to prefix that prefix to all NamespaceSelector.MatchLabels
+			for k, v := range egressRule.NamespaceSelector.MatchLabels {
+				matchLabels[policy.JoinPath(k8sConst.PodNamespaceMetaLabels, k)] = v
+			}
+			egressRule.NamespaceSelector.MatchLabels = matchLabels
+
+			// We use our own special label prefix for namespace metadata,
+			// thus we need to prefix that prefix to all NamespaceSelector.MatchLabels
+			for i, lsr := range egressRule.NamespaceSelector.MatchExpressions {
+				lsr.Key = policy.JoinPath(k8sConst.PodNamespaceMetaLabels, lsr.Key)
+				egressRule.NamespaceSelector.MatchExpressions[i] = lsr
+			}
+
+			// Empty namespace selector selects all namespaces (i.e., a namespace
+			// label exists).
+			if len(egressRule.NamespaceSelector.MatchLabels) == 0 && len(egressRule.NamespaceSelector.MatchExpressions) == 0 {
+				egressRule.NamespaceSelector.MatchExpressions = []slim_metav1.LabelSelectorRequirement{allowAllNamespacesRequirement}
+			}
+
+			endpointSelectorList = append(
+				endpointSelectorList,
+				api.NewESFromK8sLabelSelector("", prefixedNsSelector, egressRule.PodSelector))
+		} else if egressRule.PodSelector != nil {
+			endpointSelectorList = append(
+				endpointSelectorList,
+				api.NewESFromK8sLabelSelector("", egressRule.PodSelector))
+		}
+	}
+
+	return &EgressPolicyConfig{
+		uid:               cenp.UID,
+		endpointSelectors: endpointSelectorList,
+		dstCIDRs:          dstCidrList,
+		egressIp:          net.ParseIP(cenp.Spec.EgressSourceIP).To4(),
+		id: k8s.ServiceID{
+			Name: name,
+		},
+	}, nil
+}
+
+type Manager struct {
+	mutex lock.Mutex
+
+	// Stores pod to policy mapping
+	policyPods map[podID][]policyID
+	// Stores policy configs indexed by policyID
+	policyConfigs map[policyID]*EgressPolicyConfig
+	// Stores podId to pod metadata mapping
+	podDataStore map[podID]*podMetadata
+
+	storeGetter StoreGetter
+}
+
+func NewEgressPolicyManager() *Manager {
+	return &Manager{
+		policyPods:    make(map[podID][]policyID),
+		policyConfigs: make(map[policyID]*EgressPolicyConfig),
+		podDataStore:  make(map[podID]*podMetadata),
+	}
+}
+
+func (m *Manager) RegisterGetStores(sg StoreGetter) {
+	m.storeGetter = sg
+}
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "egresspolicy")
+)
+
+// Event handlers
+
+// AddEgressPolicy parses the given policy config, and updates
+// internal state with the config fields.
+func (m *Manager) AddEgressPolicy(config EgressPolicyConfig) (bool, error) {
+
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	_, ok := m.policyConfigs[config.id]
+	if ok {
+		// TODO Existing policy update
```

Comment:
Resolve this TODO.

Reply:
This is now proerly logged and err reported.